src/main/java/core/basesyntax/serviceimpl/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction: transactions) {
+            FruitTransaction.Operation operation = transaction.getOperation();
+            OperationHandler handler = operationStrategy.getHandler(operation);
+            handler.apply(transaction);
+        }
+    }
+}
```

Comment:
The 'ShopServiceImpl' class is well-encapsulated and delegates the responsibility of performing operations to the 'OperationStrategy' which is in line with the SOLID principles. However, ensure that this class is placed in the appropriate package according to the project structure. It should be under the 'service.impl' package if not already.